ui/src/app/pages/analysis/notebook-redirect.spec.tsx

```diff
@@ -0,0 +1,154 @@
+import {mount} from 'enzyme';
+import * as React from 'react';
+
+import {registerApiClient as registerApiClientNotebooks} from 'app/services/notebooks-swagger-fetch-clients';
+import {registerApiClient} from 'app/services/swagger-fetch-clients';
+import {currentWorkspaceStore, queryParamsStore, serverConfigStore, urlParamsStore, userProfileStore} from 'app/utils/navigation';
+import {Kernels} from 'app/utils/notebook-kernels';
+import {ClusterApi, ClusterStatus, WorkspaceAccessLevel} from 'generated/fetch';
+import {ClusterApi as NotebooksClusterApi, JupyterApi, NotebooksApi} from 'notebooks-generated/fetch';
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+import {ClusterApiStub} from 'testing/stubs/cluster-api-stub';
+import {JupyterApiStub} from 'testing/stubs/jupyter-api-stub';
+import {NotebooksApiStub} from 'testing/stubs/notebooks-api-stub';
+import {NotebooksClusterApiStub} from 'testing/stubs/notebooks-cluster-api-stub';
+import {ProfileStubVariables} from 'testing/stubs/profile-api-stub';
+import {workspaceStubs, WorkspaceStubVariables} from 'testing/stubs/workspaces-api-stub';
+import {NotebookRedirect} from './notebook-redirect';
+
+describe('NotebookRedirect', () => {
+  const workspace = {
+    ...workspaceStubs[0],
+    accessLevel: WorkspaceAccessLevel.OWNER,
+  };
+  const profile = ProfileStubVariables.PROFILE_STUB;
+  const reload = jest.fn();
+  const updateCache = jest.fn();
+
+  let clusterStub: ClusterApiStub;
+
+  /*
+  NotebookRedirect.progressCardStates:
```

Comment:
rm commented out code - this will inevitably go out of sync

Reply:
This was intended as a comment to convey the same information that the enum now does